extensions/ql-vscode/src/pure/location-link-utils.ts

```diff
@@ -4,8 +4,16 @@ export function createRemoteFileRef(
   fileLink: FileLink,
   startLine?: number,
   endLine?: number,
+  startColumn?: number,
+  endColumn?: number,
 ): string {
-  if (startLine && endLine) {
+  // Old GitHub code view does not show any highlighting at all if start and end line differ; so for now
+  // only add column information when the line numbers are the same
+  if (startColumn && endColumn && startLine && startLine === endLine) {
+    // Note: Currently new GitHub code view ignores column information and highlights complete lines,
```

Comment:
Also, what happens if you pass an end column number that is smaller than the start? Or what if you pass `startColumn === endColumn && startLine === endLine`?

This can be a user error if they define a bad source location in the query. It's probably harmless, and if so there's no need to handle or check for this. But, just would like to make sure.

Reply:
> Also, what happens if you pass an end column number that is smaller than the start? Or what if you pass `startColumn === endColumn && startLine === endLine`?

Good points! The old code view behaves like this:
- `startColumn === endColumn`: [No highlighting](https://github.com/github/codeql/blob/48015e5a2e6202131f2d1062cc066dc33ed69a9b/javascript/ql/test/query-tests/Security/CWE-078/tst_shell-command-injection-from-environment.js#L9C18-L9C18)
- `startColumn > endColumn`: [Highlighting](https://github.com/github/codeql/blob/48015e5a2e6202131f2d1062cc066dc33ed69a9b/javascript/ql/test/query-tests/Security/CWE-078/tst_shell-command-injection-from-environment.js#L9C18-L9C3)